packages/relay/src/lib/precheck.ts

```diff
@@ -100,7 +100,7 @@ export class Precheck {
   chainId(tx: Transaction, requestId?: string) {
     const requestIdPrefix = formatRequestIdMessage(requestId);
     const txChainId = prepend0x(Number(tx.chainId).toString(16));
-    const passes = txChainId === this.chain;
+    const passes = txChainId === '0x0' || txChainId === this.chain;
```

Comment:
Checking, do we need to support `0x1`?

Reply:
Correct me if I'm wrong but after my research, I've learned that the `chainId` didn't matter and was considered equal to 0 in legacy unprotected pre-EIP155 transactions. That's why I believe it only makes sense in this case we add support for `chainId=0x0` only if we want to allow legacy transactions. 

`0x1 chainId` indeed corresponds to the Ethereum network, and obviously Hedera doesn't support that but only the chain IDs for the 4 networks we have. 

I just tried to add support for `chainId=0x1` in `precheck.chainId()` and submitted an `eth_sendRawTransaction` with `chainId = 0x1`, then encountered an error from the SDK:
 
```bash
[2024-03-21 18:14:22.203 +0000] ERROR (consensus-node/9204 on admins-MBP-2.lan): [Request ID: [Request ID: c1a37318-b1a6-4e2f-b6ec-5dc63571c126]] Error raised during TransactionRecordQuery for 0.0.2@1711044836.808029251
    err: {
      "type": "SDKClientError",
      "message": "timeout exceeded",
      "stack":
          Error: timeout exceeded
              at SDKClient.<anonymous> (/Users/user/development/hashgraph/hedera-json-rpc-relay/packages/relay/dist/lib/clients/sdkClient.js:130:50)
              at Generator.throw (<anonymous>)
              at rejected (/Users/user/development/hashgraph/hedera-json-rpc-relay/packages/relay/dist/lib/clients/sdkClient.js:25:65)
      "status": {
        "_code": 21
      },
      "validNetworkError": false
    }
[2024-03-21 18:14:22.204 +0000] TRACE (consensus-node/9204 on admins-MBP-2.lan): [Request ID: [Request ID: c1a37318-b1a6-4e2f-b6ec-5dc63571c126]] 0.0.2@1711044836.808029251 eth_sendRawTransaction EthereumTransaction status: WRONG_CHAIN_ID (311), cost: 0
[2024-03-21 18:14:22.204 +0000] ERROR (relay-eth/9204 on admins-MBP-2.lan): [Request ID: c1a37318-b1a6-4e2f-b6ec-5dc63571c126] Failed to successfully submit sendRawTransaction for transaction 0x01f86c018085a54f4c3c00832dc6c094000000000000000000000000000000000000040c8502540be40080c001a0ad004c3e6d5c998688d077128df6e96d914836d651f1477a7e241c739651f9aca049dca929f6c9a3d7bf8188e28dae60c9cecc97e1d91f589c4d513aedbdf46ec4
    err: {
      "type": "SDKClientError",
      "message": "transaction 0.0.2@1711044836.808029251 failed precheck with status WRONG_CHAIN_ID",
      "stack":
          Error: transaction 0.0.2@1711044836.808029251 failed precheck with status WRONG_CHAIN_ID
              at SDKClient.<anonymous> (/Users/user/development/hashgraph/hedera-json-rpc-relay/packages/relay/dist/lib/clients/sdkClient.js:115:40)
              at Generator.throw (<anonymous>)
              at rejected (/Users/user/development/hashgraph/hedera-json-rpc-relay/packages/relay/dist/lib/clients/sdkClient.js:25:65)
              at processTicksAndRejections (node:internal/process/task_queues:95:5)
      "status": {
        "_code": 311
      },
      "validNetworkError": true
    }
[2024-03-21 18:14:22.204 +0000] ERROR (rpc-server/9204 on admins-MBP-2.lan): [Request ID: c1a37318-b1a6-4e2f-b6ec-5dc63571c126] Error invoking RPC: transaction 0.0.2@1711044836.808029251 failed precheck with status WRONG_CHAIN_ID
```